R/recycle.R

```diff
@@ -1,31 +1,164 @@
-#' Vector recycling
+#' Recycling rules used by r-lib and the tidyverse
+#'
+#' @description
+#' Recycling describes the concept of repeating elements of one vector to match
+#' the size of another vector. There are two rules that guide the recycling
+#' process used by packages in r-lib and the tidyverse:
+#'
+#' - Vectors of size 1 will be recycled to the size of any other vector
+#'
+#' - Otherwise, all vectors must have the same size
+#'
+#' @section Examples:
+#'
+#' Vectors of size 1 are recycled to the size of any other vector:
+#'
+#' ```
+#' tibble(x = 1:3, y = 1L)
+#' #> # A tibble: 3 × 2
+#' #>       x     y
+#' #>   <int> <int>
+#' #> 1     1     1
+#' #> 2     2     1
+#' #> 3     3     1
+#' ```
```

Comment:
```suggestion
#' ```{r}
#' tibble(x = 1:3, y = 1L)
#' ```
```

?